cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -38,6 +39,14 @@ const (
 	githubLatestReleaseLabel = "latest"
 )
 
+var (
+	// Caches used to limit the number of GitHub API calls
+
+	cacheVersions = map[string][]string{}
+	cacheReleases = map[string]*github.RepositoryRelease{}
+	cacheFiles    = map[string][]byte{}
```

Comment:
Is it worth looking into making these properties of the repositoryFactory that can then be passed into each repository, to avoid these being globals? If they weren't global it might be easier to test the caching and other functionality, WDYT?

Reply:
@JoelSpeed I made them intentionally global because this allows to share the cache between cluster-api, kubeadm-control-plane and kubeadm-bootstrap which technically are three separated providers / three different repositories, but in fact they are all reading from the same GitHub repository

Reply:
Ack 👍